packages/ember-metal/lib/meta.js

```diff
@@ -693,7 +693,17 @@ export function descriptorFor(obj, keyName, _meta) {
   let meta = _meta === undefined ? peekMeta(obj) : _meta;
 
   if (meta !== undefined) {
-    return meta.peekDescriptors(keyName);
+    let descriptor = meta.peekDescriptors(keyName);
+
+    if (descriptor === undefined && keyName in obj) {
+      // #GH16427 in case defined without `defineProperty`
+      let possibleDesc = obj[keyName];
+      if (isDescriptor(possibleDesc) && 'didChange' in possibleDesc) {
```

Comment:
'didChange' in possibleDesc doesn't belong in descriptorFor

Also we would need to move this deprecation and fallback code and make sure we don't have the fallback code elsewhere.
https://github.com/emberjs/ember.js/blob/470dddb9bbea32f0bf71219740b22c8dc8ecf68d/packages/ember-metal/lib/property_get.js#L117